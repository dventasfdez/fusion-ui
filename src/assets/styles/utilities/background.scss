.stepone-ui {
  .bg01,
  &.bg01 {
    background: var(--backgrounds-01);
  }
  .bg02,
  &.bg02 {
    background: var(--backgrounds-02);
  }
  .bg03,
  &.bg03 {
    background: var(--backgrounds-03);
  }
  .bg04,
  &.bg04 {
    background: var(--backgrounds-04);
  }
  .bgGradient,
  &.bgGradient {
    background: var(--backgrounds-gradient);
  }
  .bgAccent,
  &.bgAccent {
    background: var(--backgrounds-accent);
  }
  .bgSecondary,
  &.bgSecondary {
    background: var(--backgrounds-secondary);
  }
}
