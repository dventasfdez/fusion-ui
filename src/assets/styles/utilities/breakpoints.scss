$viewports: (
  "mobile": (
    "min-width": 0px,
    "max-width": 576px,
  ),
  "tablet": (
    "min-width": 577px,
    "max-width": 768px,
  ),
  "large": (
    "min-width": 993px,
    "max-width": 1200px,
  ),
  "extra-large": (
    "min-width": 1201px,
  ),
);

@mixin breakpoint($device) {
  @if $device == "mobile" {
    @media only screen and (min-width: 0px) and (max-width: 576px) {
      @content;
    }
  } @else if $device == "tablet" {
    @media only screen and (min-width: 577px) and (max-width: 768px) {
      @content;
    }
  } @else if $device == "large" {
    @media only screen and (min-width: 993px) and (max-width: 1200px) {
      @content;
    }
  } @else if $device == "extra-large" {
    @media only screen and (min-width: 1201px) {
      @content;
    }
  }
}
