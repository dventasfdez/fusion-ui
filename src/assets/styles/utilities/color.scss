.stepone-ui {
  .primary-color,
  &.primary-color {
    background: var(--full-palette-primary-500);
  }

  .secondary-color,
  &.secondary-color {
    background: var(--full-palette-secondary-500);
  }

  .action-color,
  &.action-color {
    background: var(--backgrounds-action-cta-enabled);
  }

  .accent-color,
  &.accent-color {
    background: var(--full-palette-accent-500);
  }

  .error-color,
  &.error-color {
    background: var(--full-palette-semantic-error-400);
  }

  .success-color,
  &.success-color {
    background: var(--full-palette-semantic-success-400);
  }

  .alert-color,
  &.alert-color {
    background: var(--full-palette-semantic-alert-400);
  }
}
