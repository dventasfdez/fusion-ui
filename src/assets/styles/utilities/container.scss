@use "sass:map";
.container {
  width: 100%;
  max-width: 1200px;
  margin: 0 auto;
}

@each $viewport, $sizes in $viewports {
  @include breakpoint($viewport) {
    .container_#{$viewport} {
      width: 100%;
      max-width: map-get($map: $sizes, $key: "max-width");
      margin: 0 auto;
    }
  }
}

.container_fluid {
  width: 100%;
  margin: 0 auto;
}
