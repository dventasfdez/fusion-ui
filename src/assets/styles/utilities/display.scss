//DISPLAYS
.display_block {
  display: block !important;
}

.display_none {
  display: none !important;
}

.display_inline {
  display: inline !important;
}

@media (max-width: $tablet) {
  .display_block_tablet {
    display: block !important;
  }

  .display_none_tablet {
    display: none !important;
  }

  .display_inline_tablet {
    display: inline !important;
  }
}

@media (max-width: $mobile) {
  .display_block_mobile {
    display: block !important;
  }

  .display_none_mobile {
    display: none !important;
  }

  .display_inline_mobile {
    display: inline !important;
  }
}
