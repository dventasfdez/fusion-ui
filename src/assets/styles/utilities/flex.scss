//FLEX
.flex {
  display: flex !important;
}

.flex_col {
  display: flex !important;
  flex-direction: column !important;
}

.flex_row {
  display: flex !important;
  flex-direction: row !important;
}

.flex_wrap {
  flex-wrap: wrap !important;
}

@include breakpoint("tablet") {
  .flex_tablet {
    display: flex !important;
  }

  .flex_col_tablet {
    display: flex !important;
    flex-direction: column !important;
  }

  .flex_row_tablet {
    display: flex !important;
    flex-direction: row !important;
  }

  .flex_wrap_tablet {
    flex-wrap: wrap !important;
  }
}

@include breakpoint("mobile") {
  .flex_mobile {
    display: flex !important;
  }

  .flex_col_mobile {
    display: flex !important;
    flex-direction: column !important;
  }

  .flex_row_mobile {
    display: flex !important;
    flex-direction: row !important;
  }

  .flex_wrap_mobile {
    flex-wrap: wrap !important;
  }
}
//ALIGN
.align-items_center {
  align-items: center !important;
}
.align-items_start {
  align-items: flex-start !important;
}
.align-items_end {
  align-items: flex-end !important;
}

@include breakpoint("tablet") {
  .align-items_center_tablet {
    align-items: center !important;
  }
  .align-items_start_tablet {
    align-items: flex-start !important;
  }
  .align-items_end_tablet {
    align-items: flex-end !important;
  }
}

@include breakpoint("mobile") {
  .align-items_center_mobile {
    align-items: center !important;
  }
  .align-items_start_mobile {
    align-items: flex-start !important;
  }
  .align-items_end_mobile {
    align-items: flex-end !important;
  }
}

//JUSTIFY CONTENT
.justify-content_center {
  justify-content: center !important;
}
.justify-content_start {
  justify-content: flex-start !important;
}
.justify-content_end {
  justify-content: flex-end !important;
}
.justify-content_between {
  justify-content: space-between !important;
}
.justify-content_around {
  justify-content: space-around !important;
}

@include breakpoint("tablet") {
  .justify-content_center_tablet {
    justify-content: center !important;
  }
  .justify-content_start_tablet {
    justify-content: flex-start !important;
  }
  .justify-content_end_tablet {
    justify-content: flex-end !important;
  }
  .justify-content_between_tablet {
    justify-content: space-between !important;
  }
  .justify-content_around_tablet {
    justify-content: space-around !important;
  }
}

@include breakpoint("mobile") {
  .justify-content_center_mobile {
    justify-content: center !important;
  }
  .justify-content_start_mobile {
    justify-content: flex-start !important;
  }
  .justify-content_end_mobile {
    justify-content: flex-end !important;
  }
  .justify-content_between_mobile {
    justify-content: space-between !important;
  }
  .justify-content_around_mobile {
    justify-content: space-around !important;
  }
}
