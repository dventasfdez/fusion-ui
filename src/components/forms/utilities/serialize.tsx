export const serializeForm = (formElement: any, excludeEmptyValues: boolean | undefined = true) => {
  // Per 17.12.1 in the HTML 4 spec:
  // Disabled controls cannot be successfully posted.
  // https://stackoverflow.com/questions/7357256/disabled-form-inputs-do-not-appear-in-the-request

  const disabledInputs = document.querySelectorAll("form :disabled");
  for (let input of disabledInputs) input.removeAttribute("disabled");
  const currentFormValues = new FormData(formElement);
  const entries: IterableIterator<[string, FormDataEntryValue]> = currentFormValues.entries();
  var object: any = {};
  // Display the key/value pairs
  for (const pair of entries) {
    const key: any = pair[0];
    const value: any = pair[1];

    if (!excludeEmptyValues || (excludeEmptyValues && value && value !== "" && typeof value != "undefined" && value !== null)) {
      processObject(object, key, value);
    }
  }
  for (let input of disabledInputs) input.setAttribute("disabled", "");
  return object;
};

function processObject(obj: any, key: any, value: any) {
  if (key && key.indexOf(".") !== -1) {
    var attrs = key.split(".");
    var tx = obj;
    for (var i = 0; i < attrs.length - 1; i++) {
      var isArray = attrs[i].indexOf("[") !== -1;
      var isNestedArray = isArray && i !== attrs.length - 1;
      var nestedArrayIndex = null;
      if (isArray) {
        nestedArrayIndex = attrs[i].substring(attrs[i].indexOf("[") + 1, attrs[i].indexOf("]"));
        attrs[i] = attrs[i].substring(0, attrs[i].indexOf("["));
        if (tx[attrs[i]] === undefined) {
          tx[attrs[i]] = [];
        }
        tx = tx[attrs[i]];
        if (isNestedArray) {
          if (tx[nestedArrayIndex] === undefined) {
            tx[nestedArrayIndex] = {};
          }
          tx = tx[nestedArrayIndex];
        }
      } else {
        if (tx[attrs[i]] === undefined) {
          tx[attrs[i]] = {};
        }
        tx = tx[attrs[i]];
      }
    }
    processObject(tx, attrs[attrs.length - 1], value);
  } else {
    var finalArrayIndex = null;
    if (key.indexOf("[") !== -1) {
      finalArrayIndex = key.substring(key.indexOf("[") + 1, key.indexOf("]"));
      key = key.substring(0, key.indexOf("["));
    }
    if (finalArrayIndex === null) {
      let existingValue = obj[key];
      if (existingValue !== value) {
        //ignore duplicated entries (like when there are 2 inputs with the same name and value inside a form)
        if (existingValue) {
          obj[key] = typeof existingValue === "object" && existingValue.constructor === Array ? existingValue : [existingValue];
          obj[key].push(value);
        } else {
          obj[key] = value;
        }
      }
    } else {
      if (obj[key] === undefined) {
        obj[key] = [];
      }
      obj[key][finalArrayIndex] = value;
    }
  }
}

export const getNestedObjectValueByString = function (object: any, s: string) {
  if (s) {
    s = s.replace(/\[(\w+)\]/g, ".$1"); // convert indexes to properties
    s = s.replace(/^\./, ""); // strip a leading dot
    var a = s.split(".");
    for (var i = 0, n = a.length; i < n; ++i) {
      var k = a[i];
      if (object) {
        if (k in object) {
          object = object[k];
        } else {
          return;
        }
      }
    }
    return object;
  }
};

export const mergeValuesIncludingEmpty = (values1: any, values2: any) => {
  var missingValues: any = {};
  if (values1) {
    for (var prop in values1) {
      if (Object.prototype.hasOwnProperty.call(values1, prop)) {
        if (!values2 || !Object.prototype.hasOwnProperty.call(values2, prop)) {
          missingValues[prop] = null;
        }
      }
    }
  }
  let newValues = { ...values2, ...missingValues };
  return newValues;
};

export const getInputValueFromEvent = (event: any) => {
  if (event && event.target) {
    let newValue = event.target.value;
    if (event.target && event.target.options) {
      var options = event.target.options;
      newValue = [];
      for (var i = 0, l = options.length; i < l; i++) {
        if (options[i].selected) {
          newValue.push(options[i].value);
        }
      }
    }
    return newValue;
  }
};
