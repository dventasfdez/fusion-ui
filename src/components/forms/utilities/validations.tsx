import { DateTime } from "luxon";

const invalidNameWithHyphenRegExp = /[\x21-\x26\x28-\x2C\x2E-\x2F\x3A-\x40\x5B-\x5D\x7B-\x7D]/;
const validationIsGood = undefined;

export const required = (value: string | string[] | undefined) => {
  if (typeof value === "number") {
    return validationIsGood;
  }
  if (typeof value === "object") {
    const value2: any = value;
    if (value2?.type === "FileSelector") {
      if (!value2?.state?.value) {
        return "validations.required";
      }
    } else if (!value?.length) {
      return "validations.required";
    }
  } else if (typeof value === "string") {
    if (value.replace(/\s/g, "").length > 0) {
      return validationIsGood;
    } else {
      return "validations.required";
    }
  } else if (value !== null && typeof value !== "undefined") {
    return validationIsGood;
  } else {
    return "validations.required";
  }
};
export const isSupportedMimeType = (input: any) => {
  const _accept = input?.props?.accept as string;
  const _current = input?.state?.fileType as string;
  return !_current || _accept?.includes(_current) ? validationIsGood : "validations.file.unsupported";
};

export const maximumFileSize = (input: any, options?: any) => {
  const maxSize = options?.maxSize || input?.props?.maximumSize;
  const fileSize = options?.fileSize || (input?.state?.fileSize as number);
  return !fileSize || fileSize < maxSize ? validationIsGood : "validations.file.tooLarge";
};

export const alphabet = (value: any) => {
  return value && /[\x21-\x26\x28-\x2F\x3A-\x40\x5B-\x5D\x7B-\x7D]/i.test(value) ? "Must be alphabet" : validationIsGood;
};
export const alphanumeric = (value: any) => {
  return value && !/^[A-Z0-9]+$/i.test(value) ? "Must be alphanumeric" : validationIsGood;
};

export const numeric = (value: any) => {
  return value && !/^[0-9]+$/i.test(value) ? "Must be numeric" : validationIsGood;
};

export const uuid = (value: any) => {
  return value && !/^[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}$/i.test(value) ? "Must be a UUID" : validationIsGood;
};

export const email = (value: any) => {
  return value && !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(value) ? "public.descriptionEmail" : validationIsGood;
};
// export const phone = (value: any) => {
//   return value && !regExpHelper.isPhone(value) ? "validations.phone.invalid"  : validationIsGood;
// };
export const _void = (_value: any) => {
  return validationIsGood;
};
export const url = (value: any) => {
  return value && !/^((https|http):\/\/)?(www\.)?(([-a-zA-Z0-9@:%._+~#=]{1,256}\.[a-zA-Z0-9()]{1,6})|localhost:3001)\b([-a-zA-Z0-9()@:%_+.~#?&/=]*)$/i.test(value)
    ? "Invalid url address"
    : validationIsGood;
};

export const hexacode = (value: any) => {
  return value && !/^#([A-F0-9]{3}|[A-F0-9]{6})$/i.test(value) ? "Invalid hexacode string" : validationIsGood;
};

export const validateAlphabetWithHyphen = (value: string) => {
  return invalidNameWithHyphenRegExp.test(value) ? "validations.invalid.character" : validationIsGood;
};
export const fiscalCode = (value: string) => {
  return value && !/^([A-Za-z]){6,6}([A-Za-z0-9]){2,2}([A-Za-z]){1,1}([A-Za-z0-9]){7,7}$/i.test(value) ? "Please enter a valid fiscal code." : validationIsGood;
};

export const date = (value: any, _context: any, inputComponent: any) => {
  if (!value || Object.prototype.toString.call(value) === "[object Date]") {
    return validationIsGood;
  } else {
    let isDate = null;
    let format = inputComponent && inputComponent.props.dateFormat ? inputComponent.props.dateFormat : null;
    if (value) {
      if (format) {
        isDate = DateTime.fromFormat(value, format).isValid;
      } else {
        isDate = DateTime.fromJSDate(value).isValid;
      }
    }
    if (isDate) {
      return validationIsGood;
    }
    const formatMessage = " (" + format + ")";
    return "Date is not correct" + formatMessage;
  }
};

export const minLength = (minLengthValue: any) => (value: any) => value && value.length < minLengthValue ? `Must be ${minLengthValue} characters or more` : validationIsGood;

export const maxLength = (maxLengthValue: any) => (value: any) => value && value.length > maxLengthValue ? `Must be ${maxLengthValue} characters or less` : validationIsGood;

export const totalLength = (totalLengthValue: any) => (value: any) =>
  value && (value.length > totalLengthValue || value.length < totalLengthValue) ? `Must be ${totalLengthValue} characters` : validationIsGood;

export const min = (minValue: any) => (value: any) => value < minValue ? `Value must be higher than ${minValue}` : validationIsGood;

export const max = (maxValue: any) => (value: any) => value > maxValue ? `Value must be lower than ${maxValue}` : validationIsGood;

export const min1 = (value: any) => (value < 1 ? `Value must be higher than 1` : validationIsGood);

export const passwordsMatch = (value: any, context: any): any => {
  const otherPwdField = context.inputs.find((a: any) => a.props.name === "password");

  return value !== otherPwdField.state.value ? "Passwords don't match" : validationIsGood;
};
export const minValidator = (value: any, allValues: any) => (value > allValues.max ? "Min cannot be higher than Max" : validationIsGood);

export const maxValidator = (value: any, allValues: any) => (value < allValues.min ? "Max cannot be lower than Min" : validationIsGood);

export const minYears = (minYearsValue: any) => (value: any, _context: any, inputComponent: any) => {
  if (value) {
    let _date = null;
    let isValidAge = false;
    let format = inputComponent && inputComponent.props.dateFormat ? inputComponent.props.dateFormat : null;
    // Get date in format
    if (format) {
      _date = DateTime.fromFormat(value, format);
    } else {
      _date = DateTime.fromJSDate(value);
    }
    // Check if is valid
    if (_date.isValid) {
      const years = DateTime.now().diff(_date, "years");
      isValidAge = years >= minYearsValue;
    }
    return !isValidAge ? `Must be ${minYearsValue} age years or more` : validationIsGood;
  }
};

export const minYears16 = minYears(16);
export const minYears18 = minYears(18);
export const minYears21 = minYears(21);

export const mustContainLowerCase = (value: any) => {
  const passwordRegex = new RegExp("^((?=.*[a-z]))");
  return value && !passwordRegex.test(value) ? "Password should contain a lower case letter" : validationIsGood;
};

export const mustContainUpperCase = (value: any) => {
  const passwordRegex = new RegExp("^((?=.*[A-Z]))");
  return value && !passwordRegex.test(value) ? "Password should contain an upper case letter" : validationIsGood;
};

export const mustContainSpecialCharacter = (value: any) => {
  // prettier-ignore
  const passwordRegex = new RegExp("^((?=.*[!@#$%^&*]))");
  return value && !passwordRegex.test(value) ? "Password should contain a Special Character" : validationIsGood;
};

export const isChecked = (input: any, context: any) => {
  if (context && context.inputs && context.inputs.length > 0) {
    return checkIfOneChecked(input, context);
  } else {
    return input && input.state && input.state.checked ? validationIsGood : "validations.required";
  }
};

export const isRadioChecked = (input: any, context: any) => {
  // Maintain this function in order to have separte logic
  // Avoid codesmell
  return isChecked(input, context);
};

function checkIfOneChecked(input: any, context: any) {
  let thereIsOneChecked = false;
  context.inputs.forEach((element: any) => {
    if (!thereIsOneChecked && element.props.name === input.props.name && element.state.checked) {
      thereIsOneChecked = true;
    }
  });
  return thereIsOneChecked ? validationIsGood : "validations.required";
}

export const hasFileDropFiles = (input: any, _context: any) => {
  return input && input.state && input.state.files && input.state.files.length ? validationIsGood : "validations.required";
};
export const hasFileUploaderFiles = (input: any, _context: any) => {
  return input && input.state && input.state.filesValue?.length > 0 ? validationIsGood : "validations.required";
};

export const hasFileSelectorFiles = (input: any) => {
  return input && input.state && input.state.value && input.state.value.length ? validationIsGood : "validations.required";
};

export interface IValidationSummaryItem {
  valid: boolean;
  inputComponent: any;
  name: string;
  value: any;
  errors: string[];
}
export interface IValidationSummary {
  valid: boolean;
  items: IValidationSummaryItem[];
}
/**
 * This function calls behind the sceenes validateInputVerbose with the difference that only returns true or false rather than an object sumarizing the errors.
 * This function goes through all validation functions passed as properties in an input, and uses them to verify the value matches all of them, if the value fails in any of those, the input would be marked as Invalid setting it's stat
 * @returns true or false
 * @param inputComponent - Is the actual input component
 * @param context - Is the form context, where the input is loaded
 */
export async function validateInput(inputComponent: any, context: any, fromFake?: boolean) {
  const validationSumaryItem: IValidationSummaryItem = await validateInputVerbose(inputComponent, context, fromFake);
  return validationSumaryItem.valid;
}

/**
 * This function goes through all validation functions passed as properties in an input, and uses them to verify the value matches all of them, if the value fails in any of those, the input would be marked as Invalid setting it's state
 * @returns an object that will contain the following
 *  IValidationSummaryItem { valid: boolean; inputComponent: any; name: string; value: any; errors: string[]; }
 * @param inputComponent - Is the actual input component
 * @param context - Is the form component, where the input is loaded
 */
export async function validateInputVerbose(inputComponent: any, context: any, fromFake?: boolean) {
  const inputValidations = inputComponent.props.validations || [];
  let validationSumaryItem: IValidationSummaryItem = {
    valid: true,
    inputComponent: inputComponent,
    name: inputComponent?.props?.name,
    value: inputComponent?.state?.value,
    errors: [],
  };
  if (inputComponent.props.visible !== false && (inputValidations?.length || inputComponent.state.requireByRelated) && !inputComponent.state.disabled) {
    let arrayOfErrors: string[] = [];
    if (inputComponent._isMounted) {
      inputComponent.setState({ loading: true });
    }
    // Required related
    if (inputComponent.state.requireByRelated) {
      const errorRelatedRequired = required(inputComponent.state.value);
      if (typeof errorRelatedRequired !== "undefined") arrayOfErrors.push("validations.required.related");
    }
    for await (const validationFunction of inputValidations) {
      if (typeof validationFunction === "function") {
        const type = inputComponent.state.type;
        let errorMessage: string | undefined = "";
        switch (type) {
          case "checkbox":
            if (validationFunction === required) errorMessage = isChecked(inputComponent, context);
            else errorMessage = await validationFunction(inputComponent.state.value, context, inputComponent, fromFake);
            break;
          case "radio":
            if (validationFunction === required) errorMessage = isRadioChecked(inputComponent, context);
            else errorMessage = await validationFunction(inputComponent.state.value, context, inputComponent, fromFake);
            break;
          case "FileDrop":
            errorMessage = hasFileDropFiles(inputComponent, context);
            break;
          case "FileUploader":
            errorMessage = hasFileUploaderFiles(inputComponent, context);
            break;
          case "FileSelector": //add required validator if content is mandatory
            errorMessage = validationFunction(inputComponent);
            break;
          default:
            errorMessage = await validationFunction(inputComponent.state.value, context, inputComponent, fromFake);
        }

        if (typeof errorMessage !== "undefined") {
          arrayOfErrors.push(errorMessage);
        }
        validationSumaryItem.errors = arrayOfErrors;
      }
    }
    if (arrayOfErrors && arrayOfErrors.length) {
      if (inputComponent._isMounted) {
        inputComponent.setState({
          errors: arrayOfErrors,
          isValid: false,
          loading: false,
        });
      }
      validationSumaryItem.valid = false;
      return validationSumaryItem;
    } else {
      if (inputComponent._isMounted) {
        inputComponent.setState({ errors: null, isValid: true, loading: false });
      }
      return validationSumaryItem;
    }
    // If not validation is valid
  } else {
    if (inputComponent._isMounted) {
      inputComponent.setState({ isValid: true });
    }
    return validationSumaryItem;
  }
}
