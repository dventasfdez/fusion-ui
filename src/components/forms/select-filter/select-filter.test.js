import React from "react";
import renderer from "react-test-renderer";
import ReactDOM from "react-dom";
import ReactTestUtils from "react-dom/test-utils";
import SelectFilter, { Option } from "./select-filter";
import { required } from "../forms/utilities/validations";

const SelectFilterExample = () => {
  return (
  <div>
     <SelectFilter className="mb2" tooltip="tooltip example" label="Select Colour (multiple)" validations={[required]} value={["1", "5"]} multiple={true} name="color">
       <Option value="1" label="Red" />
       <Option value="2" label="Yellow" />
        <Option value="3" label="Orange" />
       <Option value="4" label="Pink" />
       <Option value="5" label="Purple" />
       <Option value="6" label="Blue" />
       <Option value="7" label="Grey" />
       <Option value="8" label="White" />
       <Option value="9" label="Black" />
     </SelectFilter>

    </div>
  );
};

test("SelectFilter should render and match snapshot", () => {
  ReactDOM.createPortal = jest.fn((dummy) => dummy); //overrides the create portan function
  const component = renderer.create(<SelectFilterExample />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


