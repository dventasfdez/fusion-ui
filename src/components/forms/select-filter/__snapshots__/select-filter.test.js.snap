// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectFilter should render and match snapshot 1`] = `
<div>
  <div
    className="input-wrapper  validation-required   mb2"
  >
    <label
      className="caption mb1"
    >
      <small
        className="required"
      >
        *
      </small>
      Select Colour (multiple)
      <span
        className="material-icons info-tooltip ml1"
      >
        info
      </span>
    </label>
    <div
      className="hidden"
    >
      <select
        multiple={true}
        name="color"
        onChange={[Function]}
        value={
          Array [
            "1",
            "5",
          ]
        }
      >
        <option
          value="1"
        >
          Red
        </option>
        <option
          value="2"
        >
          Yellow
        </option>
        <option
          value="3"
        >
          Orange
        </option>
        <option
          value="4"
        >
          Pink
        </option>
        <option
          value="5"
        >
          Purple
        </option>
        <option
          value="6"
        >
          Blue
        </option>
        <option
          value="7"
        >
          Grey
        </option>
        <option
          value="8"
        >
          White
        </option>
        <option
          value="9"
        >
          Black
        </option>
      </select>
    </div>
    <div
      className="dropdown "
      data-show={false}
    >
      <div
        className="dropdown-button "
        onClick={[Function]}
      >
        <div
          className="input-container"
        >
          <input
            className="select-text-input"
            data-testid="select-filter"
            onChange={[Function]}
            placeholder="Click to search"
            type="text"
            value=""
          />
          <span
            className="material-icons input-icon-box"
          >
            search
          </span>
        </div>
      </div>
    </div>
    <div
      className="select-bottom-container multiple-element"
    >
      <button
        className="chip_active"
        id="selected-1"
        onClick={[Function]}
        type="button"
      >
        Red
        <span
          className="material-icons right"
        >
          close
        </span>
      </button>
      <button
        className="chip_active"
        id="selected-5"
        onClick={[Function]}
        type="button"
      >
        Purple
        <span
          className="material-icons right"
        >
          close
        </span>
      </button>
    </div>
  </div>
</div>
`;
